eva.py

```diff
@@ -199,8 +198,69 @@ def crop_edge_times(da, drop):
     return da.isel({'time': slice(start, stop)})
 
 
+def fix_output_metadata(da, mode, freq, extreme_type, outvar=None, season=None, month=None):
+    """Edit xclim default output variable attributes.
+
+    Parameters
+    ----------
+    da : xarray DataArray
+    mode : {'min', 'max'}
+        Look for probability of exceedance (max) or non-exceedance (min)
+    freq : str
+        Time frequency for blocks
+    extreme_type : {'ari', 'aep', 'quantile'}
+        Type of extreme to return
+        ari = average recurrence interval
+        aep = annual exceedance probability
+    outvar : str, optional
+        Name of custom output variable
+    month : list, optional
+        Restrict analysis to these months (list of month numbers)
+    season : {'DJF', 'MAM', 'JJA', 'SON'}, optional
+        Restrict analysis to a particular season
+
+    """
+
+    del da.attrs['cell_methods']
+    del da.attrs['method']
+    del da.attrs['history']
+    da.attrs['long_name'] = da.attrs['original_long_name']
+    del da.attrs['original_long_name']
+
+    da.attrs['block_frequency'] = freq
+    extreme_type_text = extreme_type.upper() if extreme_type in ['ari', 'aep'] else extreme_type
+    da.attrs['description'] = da.attrs['description'].replace('Quantile', extreme_type_text)
+    if mode == 'min':
+        da.attrs['direction'] = 'probability of non-exceedance'
+    elif mode == 'max':
+        da.attrs['direction'] = 'probability of exceedance'
+    if season:
+        da.attrs['season'] = season
+    if month:
+        da.attrs['months'] = str(month)
+
+    if outvar:
+        da = da.rename(outvar)
+        if outvar in ['txx', 'tnx']:
+            da.attrs['long_name'] = 'Maxima of ' + da.attrs['long_name']
+        elif outvar in ['txn', 'tnn']:
+            da.attrs['long_name'] = 'Minima of ' + da.attrs['long_name']
```

Comment:
@tonyrafter If the user uses the new `--outvar` option at the command line two things happen:
1. The output variable takes on the name they specify
2. If the name they specify appears in this if statement the `long_name` attribute is updated accordingly (you could add more names to this if statement over time if you wanted)

Does that sound like a good solution?

Reply:
That's brilliant, thanks Damien. 

Initially I was going to say that I would add something for precip. A little more complicated though as the extreme in question will depend on the duration - e.g. 1-day max precip is `rx1day`, 5-day max is `rx5day` etc. 

**However,** perhaps an easier / more elegant solution would just be to add a prefix to each `long_name`, either `'Maxima of ' + da.attrs['long_name']`  or `'Minima of ' + da.attrs['long_name']` depending on the value given for `mode`? That way whatever variable is supplied will just take the original `long_name` and adjust it for the `mode`, regardless of what the output variable name is. 

Can you see any problem with doing it this way? I may be missing something...

Side note: It might be worth adding a check that it is from a list of known extreme variables e.g. `txx`, `tnn`, `rx1day` etc., noting though that if you want to do extremes of a non-standard variable (e.g. FFDI) there isn't a defined extreme name for it. So, maybe not?!

Reply:
Linking the long name to the mode makes much more sense (I've made that update to the script).

Thinking about it a little more, I think we probably want to steer away of trying to maintain a list of known extremes variables. Maintaining that list would be a bit annoying.